core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -481,13 +487,20 @@ object H2OContext extends Logging {
     * @return H2O Context
     */
   def getOrCreate(sparkSession: SparkSession, conf: H2OConf): H2OContext = synchronized {
-    if (instantiatedContext.get() == null) {
-      val isNonJVMClient = conf.getBoolean(SharedBackendConf.PROP_RUNNING_FROM_NON_JVM_CLIENT._1,
-        SharedBackendConf.PROP_RUNNING_FROM_NON_JVM_CLIENT._2)
-      val isExternalBackend = conf.runsInExternalClusterMode
-      if (isExternalBackend && isNonJVMClient) {
-        instantiatedContext.set(new H2OContextRestAPIBased(sparkSession, conf).init())
+    val isNonJVMClient = conf.getBoolean(SharedBackendConf.PROP_RUNNING_FROM_NON_JVM_CLIENT._1,
+      SharedBackendConf.PROP_RUNNING_FROM_NON_JVM_CLIENT._2)
+    val isExternalBackend = conf.runsInExternalClusterMode
+    if (isExternalBackend && isNonJVMClient) {
+      if (instantiatedContext.get() != null) {
+        if (connectingToNewCluster(instantiatedContext.get(), conf)) {
```

Comment:
Could happen that the `connectingToNewCluster` method is called when the external cloud is still clouding up?

Reply:
yup, but I think it is really rare case. After H2O node is started, one of the first things it does is that it starts listening for the incoming requests. So even though the cloud is not yet fully formed, this call will succeed but instead of showing all nodes, just nodes available
 at the time of the call will be printed out. 

Which I think is fine in the manual mode as the use-case there is to have cluster running to which we want to connect

Reply:
Ok, perfect. Thanks!